src/components/SQForm/SQForm.tsx

```diff
@@ -1,25 +1,55 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import Grid from '@material-ui/core/Grid';
 import {Formik, Form} from 'formik';
 import {useDebouncedCallback} from 'use-debounce';
 import {object as YupObject, setLocale} from 'yup';
 import {useInitialRequiredErrors} from '../../hooks/useInitialRequiredErrors';
+import type {GridProps} from '@material-ui/core';
+import type {FormikHelpers, FormikValues} from 'formik';
+import type {ObjectShape} from 'yup/lib/object';
 
 setLocale({
   mixed: {
     required: 'Required',
   },
 });
 
-function SQForm({
+interface SQFormProps<Values> {
+  /** Form Input(s) */
+  children: React.ReactNode;
+  /** Bool to pass through to Formik. https://formik.org/docs/api/formik#enablereinitialize-boolean */
+  enableReinitialize?: boolean;
+  /** Form Entity Object */
+  initialValues: Record<string, Values>;
+  /** Any prop from https://material-ui.com/api/grid */
+  muiGridProps?: GridProps;
+  /**
+   * Form Submission Handler | @typedef onSubmit: (values: Values, formikBag: FormikBag) => void | Promise<any>
+   * IMPORTANT: If onSubmit is async, then Formik will automatically set isSubmitting to false on your behalf once it has resolved.
+   * This means you do NOT need to call formikBag.setSubmitting(false) manually.
+   * However, if your onSubmit function is synchronous, then you need to call setSubmitting(false) on your own.
+   *
+   * https://jaredpalmer.com/formik/docs/api/withFormik#handlesubmit-values-values-formikbag-formikbag--void--promiseany
+   * */
+  onSubmit: (
```

Comment:
I think we should change `onSubmit` to be optional with this TS conversion. We have the use case where SQForm is used for read-only forms and thus `onSubmit` isn't used. I'd be interested to hear thoughts on this, though.

Reply:
Yeah I'm good with that as well now that I think about it.  I have personally had multiple times I have passed `onSubmit` an empty function `(() => {})` just to make `SQForm` happy.

Reply:
I am curious for additional thoughts on this.  It seems to me that it is legitimate to make the `onSubmit` prop optional, but would like some additional opinions before I make this change.

Reply:
Can confirm I've had to send in noops to onSubmit several times before and agree making it optional makes sense.

Reply:
Sounds good @caleboleary.  I will make the change.  Thanks for the input.